### ISSUE TITLE
Issues with setting max_bval

### ISSUE THREAD

**Issue by audreycluo** [yeatmanlab/pyAFQ#1156]
Hi team! I've been trying to run pyAFQ while excluding the highest bvalue. I've updated my optional arguments from having `"max_bval": ""` to the following:
```
"min_bval": "",
"max_bval": 1500, 
"filter_b": true,
"robust_tensor_fitting": false,
"csd_response": "",
```
After changing my `max_bval`, I'm getting this error:
```
WARNING:AFQ:Failed to export warped b0. This could be because your mapping type is only compatible with transformation from template to subject space. 
The error is: boolean index did not match indexed array along dimension 3; dimension is 398 but corresponding boolean dimension is 172
```
For context, I've run pyAFQ through qsiprep 0.22.0, which includes pyAFQ optional arguments as part of a json file, and things have worked beautifully before. In the steps before pyAFQ, I use mrtrix msmt csd and tractography with all my b-values. Now that I'm excluding the max bval, I’m wondering if this is tripping up the input into pyAFQ somehow? I assume I should be able to use all the available b-values for mrtrix, then exclude my highest b-values for pyafq right?

**Comment by arokem** [comment]
My guess, based on this error, is that we have a bug that has to do with the b-value threshold being applied in some places, but not in others, so you get this kind of mismatch here. Unfortunately, I am not going to have time to look into this more closely before Friday at the very earliest.

**Comment by smeisler** [comment]
I am going to guess that it comes from the definition of `mapping` (https://github.com/yeatmanlab/pyAFQ/blob/cef7b1f9692f121184077e25b189648cb415263f/AFQ/tasks/mapping.py#L74) which is used in the `export_registered_b0` function  (https://github.com/yeatmanlab/pyAFQ/blob/cef7b1f9692f121184077e25b189648cb415263f/AFQ/tasks/mapping.py#L25).

```
def mapping(base_fname, dwi_data_file, reg_subject, data_imap,
            mapping_definition=None):
    """
    mapping from subject to template space.

    Parameters
    ----------
    mapping_definition : instance of `AFQ.definitions.mapping`, optional
        This defines how to either create a mapping from
        each subject space to template space or load a mapping from
        another software. If creating a map, will register reg_subject and
        reg_template.
        If None, use SynMap()
        Default: None
    """
    reg_template = data_imap["reg_template"]
    if mapping_definition is None:
        mapping_definition = SynMap()
    if not isinstance(mapping_definition, Definition):
        raise TypeError(
            "mapping must be a mapping defined"
            + " in `AFQ.definitions.mapping`")
    return mapping_definition.get_for_subses(
        base_fname, data_imap["dwi"], dwi_data_file,
        reg_subject, reg_template)

```

Perhaps the DWI from `data_imap["dwi"]` becomes mismatched with `dwi_data_file` when the b_val thresholding is applied?

**Comment by 36000** [comment]
Yes, this is correct, in fact, `dwi_data_file` probably should not be used anywhere in the pimms system outside of its initial use where are the bval selections happen and the orientation happens. Afterwards it should just be `dwi`. I am going to review and change uses of them, then open a PR, and we'll see if that fixes it

**Comment by 36000** [comment]
I was able to reproduce and fix the bug. Its not in the mapping code as it just uses that file path as a reference for BIDS to find other files. But it was a similar idea in the calculation of the b0. Here is the PR, if it passes all tests we can merge it: #1158

**Comment by arokem** [comment]
I believe this is closed by #1158, but please reopen if you see it again.
